<?xml version="1.0"?>
<!-- vim: set fenc=utf-8 et sts=4 ft=xml : -->
<launch>
    <arg name="bagfile"/>

    <param name="use_sim_time" type="bool" value="true"/>

    <node
        pkg="rosbag"
        type="play"
        name="replay"
        args="--pause --clock --loop
            --topics /cameras/navigation/center/left/image_raw
                    /cameras/navigation/center/left/camera_info
                    /cameras/navigation/center/right/image_raw
                    /cameras/navigation/center/right/camera_info
                    tf
            --bags $(arg bagfile)"
        output="screen">
        <remap from="/cameras/navigation/center/left/image_raw"
               to="left/image_raw"/>
        <remap from="/cameras/navigation/center/right/image_raw"
               to="right/image_raw"/>
        <remap from="/cameras/navigation/center/left/camera_info"
               to="left/camera_info"/>
        <remap from="/cameras/navigation/center/right/camera_info"
               to="right/camera_info"/>
     </node>

     <!--
    <node
        pkg="nodelet"
        type="nodelet"
        name="manager"
        args="manager"
        output="screen"/>
    -->

    <!-- Debayer -->
    <node
        pkg="nodelet"
        type="nodelet"
        name="debayer"
        args="load image_proc/debayer /manager"
        ns="left"
        output="screen">
    </node>
    <node
        pkg="nodelet"
        type="nodelet"
        name="debayer"
        args="load image_proc/debayer /manager"
        ns="right"
        >
    </node>

    <node
        pkg="nodelet"
        type="nodelet"
        name="stereo"
        args="load gpuimageproc/Stereoproc manager"
        >
        <param name="approximate_sync" type="bool" value="true"/>
    </node>

</launch>
